### PR TITLE
New fusion360 Cheat Sheet

### PR DESCRIPTION
**Pull request by salmanfarisvp**
## Description of new Instant Answer, or changes
fusion360 keyboard shortcut

## Related Issues and Discussions
Source: https://www.cadlinecommunity.co.uk/hc/en-us/articles/210361725-Fusion-360-Keyboard-Shortcuts

## Testing & Review
To be completed by Community Leader (or DDG Staff) when reviewing Pull Request

**Pull Request**
- [ ] Title follows correct format (Specifies Instant Answer + Purpose)
- [ ] Description contains a valid Instant Answer Page Link (e.g. https://duck.co/ia/view/my_ia)

**Instant Answer Page** (for new Instant Answers)
- [ ] Instant Answer page is correctly filled out and contains:
    - The **Title** is appropriately named and formatted
    - The **IA topics** are present and appropriate
    - The **Description** is clear and coherent
    - **Source Name** exists if applicable
    - All **Example Queries** trigger on [Beta](https://beta.duckduckgo.com/)

**Code**
- [ ] Adheres to the [DuckDuckGo Style Guide](https://docs.duckduckhack.com/resources/code-style-guide.html)
- [ ] Behaviour is appropriately tested. If improvement, tests are adequately extended.
- [ ] There is no unnecessary files in place (such as editor config files)
- [ ] There is no API keys / secrets present
- [ ] Tests are passing (run `$ duckpan test <goodie_id>`)
    - Tester should report any failures

**Ready to merge?**

- [ ] Has this IA been deployed to and tested on [beta.duckduckgo.com](https://beta.duckduckgo.com/)?
- [ ] For larger commits, has this been approved be more than one community member?
- [ ] The number of reviews is appropriate for this type of PR
- [ ] The commit message is clear, coherent and fitting

**Pull Request Review Guidelines**: https://docs.duckduckhack.com/programming-mission/pr-review.html

---

Instant Answer Page: https://duck.co/ia/view/fusion360_keyboard_shortcuts_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by salmanfarisvp** [comment]
now the id is `fusion360_keyboard_shortcuts_cheat_sheet` but same build error

File: share/goodie/cheat_sheets/json/fusion360.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
     "id": "fusion360_keyboard_shortcuts_cheat_sheet",
```
Comment (kirkins) [inline share/goodie/cheat_sheets/json/fusion360.json]: If the file name is fusion360 the ID would be fusion360_cheat_sheet.

They need to match.

**Comment by salmanfarisvp** [comment]
Thanks @kirkins ,

**Comment by kirkins** [comment]
@salmanfarisvp if you want to update WhatsApp too you should do it as a separate PR.

Do this to get back on track:

1. Soft reset to commit 6559231, before you added WhatsApp. 
2. Checkout master, then checkout a new branch for WhatsApp.
3. Commit WhatsApp changes to a new branch isolated from Fusion360 changes.
4. Make PR request for WhatsApp changes.

File: share/goodie/cheat_sheets/json/whatsapp.json [hunk 2]
```diff
@@ -1,97 +1 @@
...
-            },
-            {
-                "key": "Alt+F4",
-                "val": "Close"
-            }
-       
-        ]
-    }
-}
+
```
Comment (kirkins) [inline share/goodie/cheat_sheets/json/whatsapp.json]: File still exists here.

**Comment by jophab** [comment]
@salmanfarisvp CheatSheets' PR should contain only one file.
Lets remove Whatsapp.json file from this PR.

**Comment by mintsoft** [comment]
@salmanfarisvp ping! Are you OK with the changes here?

**Comment by salmanfarisvp** [comment]
@mintsoft can please check now ...

**Comment by kirkins** [comment]
@salmanfarisvp you need to fix your tests first:

```
# Testing fusion360...................... 1 WARNING, 2 FAILURES
# 	 -> FAIL: The following sections were defined in 'section_order' but not used in 'sections': ('Mouse/Touchpad')
# 	 -> FAIL: The following sections were used in 'sections' but not defined in 'section_order:' ('Mosue/Touchpad')
# 	 -> WARN: 
# Cheat sheet with ID "fusion360_cheat_sheet" not found in metadata.  To be included, you'll need to:
# 
#     1. Create an instant answer page (https://duck.co/ia/new_ia)
#     2. Open a pull request on GitHub (http://docs.duckduckhack.com/submitting/pull-request.html)
# 
# #2 is necessary to have the IA Page move from the "planning" stage to "development" automatically.
# 
# 
#   Failed test 'FAIL: The following sections were used in 'sections' but not defined in 'section_order:' ('Mosue/Touchpad')'
#   at t/CheatSheets/CheatSheetsJSON.t line 288.
```

File: share/goodie/cheat_sheets/json/fusion360.json [hunk 3]
```diff
@@ -0,0 +1,349 @@
...
+        "sourceName": "fusion360"
+    },
+    "template_type": "keyboard",
+    "section_order": [
+        "System",
+        "General",
+        "Model_workspace",
+        "Sculpt_workplace",
+        "CAM_workspace",
+        "Mouse/Touchpad"
```
Comment (kirkins) [inline share/goodie/cheat_sheets/json/fusion360.json]: One test failure is related to this not being used. Remove it if it is empty.

File: share/goodie/cheat_sheets/json/fusion360.json [hunk 4]
```diff
@@ -0,0 +1,349 @@
+{
+    "id": "fusion360_cheat_sheet",
```
Comment (kirkins) [inline share/goodie/cheat_sheets/json/fusion360.json]: Change the IA name for your IA page: https://duck.co/ia/view/fusion360_keyboard_shortcuts_cheat_sheet

You have it as ID `fusion360_keyboard_shortcuts_cheat_sheet` in one spot and `fusion360_cheat_sheet` in another.

I'd go with the second.

File: share/goodie/cheat_sheets/json/fusion360.json (hunk 3 above)
Comment (salmanfarisvp) [reply share/goodie/cheat_sheets/json/fusion360.json]: empty space...? can you please explain

Comment (salmanfarisvp) [reply share/goodie/cheat_sheets/json/fusion360.json]: thanks , can you check now .

**Comment by lotsathought** [comment]
submitted by both of my flaws? internal and my perspective is a couple

File: share/goodie/cheat_sheets/json/fusion360.json [hunk 5]
```diff
@@ -0,0 +1,353 @@
+{
+    "id": "fusion360_keyboard_shortcuts_cheat_sheet",
```
Comment (kirkins) [inline share/goodie/cheat_sheets/json/fusion360.json]: ID and file name don't match

Comment (kirkins) [reply share/goodie/cheat_sheets/json/fusion360.json]: # Testing fusion360...................... 2 FAILURES
# 	 -> FAIL: Invalid file name (fusion360.json) for ID (fusion360_keyboard_shortcuts_cheat_sheet)
# 	 -> FAIL: Uppercase detected in alias - aliases should be lowercase
    #   Failed test 'FAIL: Uppercase detected in alias - aliases should be lowercase'
    #   at t/CheatSheets/CheatSheetsJSON.t line 288.
    # Looks like you failed 1 test of 1.
#   Failed test 'can_build'
#   at t/CheatSheets/CheatSheetsJSON.t line 289.

File: share/goodie/cheat_sheets/json/fusion360.json [hunk 6]
```diff
@@ -0,0 +1,353 @@
...
+    "name": "fusion360",
+    "description": "Fusion360 Keyboard shortcuts",
+    "metadata": {
+        "sourceName": "fusion360",
+        "sourceUrl":"https://www.cadlinecommunity.co.uk/hc/en-us/articles/210361725-Fusion-360-Keyboard-Shortcuts"
+    },
+    "aliases": [
+        "Fusion 360"
```
Comment (kirkins) [inline share/goodie/cheat_sheets/json/fusion360.json]: Make lowercase f

**Comment by pjhampton** [comment]
Hey @salmanfarisvp 

[DuckDuckHack is now in Maintenance Mode](https://duckduckhack.com/) and from now on, we are only accepting issues and PRs for essential bugs and bug fixes.

Unfortunately, this is non-essential and will be closed.

We appreciate you taking the time to contribute and apologize for not being able to include your work.

**Pull request closed without merging**